packages/iTwinUI-react/src/core/Dialog/DialogMain.tsx

```diff
@@ -0,0 +1,149 @@
+ *     </Button>
+ *   </Dialog.ButtonBar>
+ * </Dialog.Main>
+ */
+export const DialogMain = React.forwardRef<HTMLDivElement, DialogMainProps>(
+  (props, ref) => {
+    const dialogContext = useDialogContext();
+    const {
+      className,
+      children,
+      styleType = 'default',
+      isOpen = dialogContext.isOpen,
+      isDismissible = dialogContext.isDismissible,
+      onClose = dialogContext.onClose,
+      closeOnEsc = dialogContext.closeOnEsc,
+      trapFocus = dialogContext.trapFocus,
+      preventDocumentScroll = dialogContext.preventDocumentScroll,
+      onKeyDown,
+      ...rest
+    } = props;
+
+    useTheme();
+
+    const dialogRef = React.useRef<HTMLDivElement>(null);
+    const refs = useMergedRefs(dialogRef, ref);
+
+    // Focuses dialog when opened and brings back focus to the previously focused element when closed.
+    const previousFocusedElement = React.useRef<HTMLElement | null>();
+    React.useEffect(() => {
+      if (isOpen) {
+        previousFocusedElement.current = dialogRef.current?.ownerDocument
+          .activeElement as HTMLElement;
+        dialogRef.current?.focus();
+      } else {
+        previousFocusedElement.current?.focus();
+      }
+      const ref = dialogRef.current;
+      return () => {
+        ref?.contains(document.activeElement) &&
+          previousFocusedElement.current?.focus();
+      };
+    }, [isOpen]);
+
+    // Prevents document from scrolling when the dialog is open.
+    const originalBodyOverflow = React.useRef('');
+    React.useEffect(() => {
+      const ownerDocument = dialogRef.current?.ownerDocument;
+      if (!ownerDocument || !preventDocumentScroll) {
+        return;
+      }
+
+      if (isOpen) {
+        originalBodyOverflow.current = ownerDocument.body.style.overflow;
+        ownerDocument.body.style.overflow = 'hidden';
+      } else {
+        ownerDocument.body.style.overflow = originalBodyOverflow.current;
+      }
+      return () => {
+        ownerDocument.body.style.overflow = originalBodyOverflow.current;
+      };
+    }, [isOpen, preventDocumentScroll]);
+
+    const handleKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+      // Prevents React from resetting its properties
+      event.persist();
+      if (isDismissible && closeOnEsc && event.key === 'Escape' && onClose) {
+        onClose(event);
+      }
+      onKeyDown?.(event);
+    };
+
+    const content = (
+      <div
+        className={cx(
+          'iui-dialog',
+          {
+            'iui-dialog-default': styleType === 'default',
+            'iui-dialog-full-page': styleType === 'fullPage',
+            'iui-dialog-visible': isOpen,
+          },
+          className,
+        )}
+        role='dialog'
+        ref={refs}
+        onKeyDown={handleKeyDown}
+        tabIndex={-1}
+        {...rest}
+      >
+        {children}
+      </div>
+    );
+
+    return (
+      <CSSTransition
+        in={isOpen}
+        classNames='iui-dialog-animation'
```

Comment:
Maybe I'm missing something but this animation doesn't work anymore because it is trying to apply the class to first child it finds (which is the dummy `<div>` placed by FocusTrap). I think it will work if you move the CSSTransition to wrap the `iui-dialog` element.

Reply:
Actually that will also not work because FocusTrap needs a ref from the child, and CSSTransition will not provide that.

Reply:
Maybe we should convert the FocusTrap component into a hook that returns a callback ref which we can pass to `<div class='iui-dialog'>`

Reply:
Have you tested it live? Because everything seems to work for me. �
![dialog-focus-animation](https://user-images.githubusercontent.com/36186912/181465894-77b6231c-bdc8-4a20-a535-a0a14bff7287.gif)

Reply:
I did a fresh clone just for you and recorded a video showing the broken animation and the class being applied on the wrong element.

https://user-images.githubusercontent.com/9084735/181819183-4d5dec0f-60a7-45a5-8e79-6e9a56df1280.mov

Reply:
Oh, on full page it is more obvious. I found a prop on CssTransitionGroup that helped here - `nodeRef`. Now it applies class to correct div:
![image](https://user-images.githubusercontent.com/36186912/182094776-da0da66f-75c3-4470-8a29-1aaef96d64f3.png)